linera-base/src/identifiers.rs

```diff
@@ -121,6 +124,44 @@ impl<A> Clone for BytecodeId<A> {
 
 impl<A> Copy for BytecodeId<A> {}
 
+impl<A> PartialEq for BytecodeId<A> {
+    fn eq(&self, other: &Self) -> bool {
+        self.message_id == other.message_id
+    }
+}
+
+impl<A> Eq for BytecodeId<A> {}
+
+impl<A> PartialOrd for BytecodeId<A> {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.message_id.partial_cmp(&other.message_id)
+    }
+}
+
+impl<A> Ord for BytecodeId<A> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.message_id.cmp(&other.message_id)
+    }
+}
+
+impl<A> Hash for BytecodeId<A> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.message_id.hash(state);
+    }
+}
+
+impl<A> fmt::Debug for BytecodeId<A> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "BytecodeId {{ message_id: {} }}", self.message_id)
+    }
+}
+
+impl<A> fmt::Display for BytecodeId<A> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "BytecodeId {{ message_id: {} }}", self.message_id)
```

Comment:
I wonder if we should show the BCS bytes (in hexadecimal) here instead, @ma2bd?

Reply:
Isn't it implemented somewhere else already? (bcs_scalar maybe?)

Reply:
Ah, right. Then this shouldn't even compile and can be removed. :+1: